src/WebApp/Extensions/Extensions.cs

```diff
@@ -91,17 +94,22 @@ public static void AddAuthenticationServices(this IHostApplicationBuilder builde
     private static void AddAIServices(this IHostApplicationBuilder builder)
     {
         var openAIOptions = builder.Configuration.GetSection("AI").Get<AIOptions>()?.OpenAI;
-        if (!string.IsNullOrWhiteSpace(openAIOptions?.ApiKey))
+        var deploymentName = openAIOptions?.ChatModel;
+
+        if (!string.IsNullOrWhiteSpace(deploymentName))
         {
-            var kernelBuilder = builder.Services.AddKernel();
-            if (!string.IsNullOrWhiteSpace(openAIOptions.Endpoint))
-            {
-                kernelBuilder.AddAzureOpenAIChatCompletion(openAIOptions.ChatModel, openAIOptions.Endpoint, openAIOptions.ApiKey);
-            }
-            else
+            builder.Services.AddKernel();
+            builder.AddAzureOpenAI("chatOpenAi");
+
+            // Note: SemanticKernel should expose an extension method AddAzureOpenAIChatCompletion() that does the following.
+            // Currently these methods only allow to register an OpenAIClient they construct.
+            Func<IServiceProvider, object?, AzureOpenAIChatCompletionService> implementationFactory = delegate (IServiceProvider serviceProvider, object? _)
             {
-                kernelBuilder.AddOpenAIChatCompletion(openAIOptions.ChatModel, openAIOptions.ApiKey);
-            }
+                return new AzureOpenAIChatCompletionService(deploymentName, serviceProvider.GetRequiredService<OpenAIClient>(), null, serviceProvider.GetService<ILoggerFactory>());
+            };
+
+            builder.Services.AddKeyedSingleton(default(string), (Func<IServiceProvider, object, IChatCompletionService>)implementationFactory);
+            builder.Services.AddKeyedSingleton(default(string), (Func<IServiceProvider, object, ITextGenerationService>)implementationFactory);
```

Comment:
The second line here isn't needed. The app only uses the IChatCompletionService.